File: 0683/Sesión 04 - lib java/Publicaciones_v03/src/main/java/pe/edu/pucp/inf30/sesiones/publicaciones03/publicaciones/AbstractFactoryPublicacion.java
package pe.edu.pucp.inf30.sesiones.publicaciones03.publicaciones;

public interface AbstractFactoryPublicacion {
    
    public abstract Publicacion crearArticulo(String nombre, String nombre_completo_autor, Integer año, String revista, Integer volumen, Integer numero, String mes);
    
    public abstract Publicacion crearArticulo(String nombre, String nombres_completos_autores[], Integer año, String revista, Integer volumen, Integer numero, String mes);
    
    public abstract Publicacion crearLibro(String nombre, String nombre_completo_autor, Integer año, String edicion, String lugar, String editorial);
    
    public abstract Publicacion crearLibro(String nombre, String nombres_completos_autores[], Integer año, String edicion, String lugar, String editorial);
    
    public abstract Publicacion crearTesis(String nombre, String nombre_completo_autor, Integer año, Clase_de_Tesis clase, String departamento, String institucion, String ciudad, String estado);
    
    public abstract Publicacion crearTesis(String nombre, String nombres_completos_autores[], Integer año, Clase_de_Tesis clase, String departamento, String institucion, String ciudad, String estado);
}

File: 0682/Sesión 05 - lib C#/Publicaciones_v03/Publicaciones_v03/Publicaciones/AbstractFactoryPublicacion.cs
using Publicaciones;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Publicaciones_v01.Publicaciones
{
    public interface AbstractFactoryPublicacion
    {
        Publicacion CrearArticulo(string nombre, string nombre_completo_autor, int año, string revista, int volumen, int numero, string mes);

        Publicacion CrearArticulo(string nombre, string[] nombres_completos_autores, int año, string revista, int volumen, int numero, string mes);

        Publicacion CrearLibro(string nombre, string nombre_completo_autor, int año, string edicion, string lugar, string editorial);

        Publicacion CrearLibro(string nombre, string[] nombres_completos_autores, int año, string edicion, string lugar, string editorial);

        Publicacion CrearTesis(string nombre, string nombre_completo_autor, int año, Clase_de_Tesis clase, string departamento, string institucion, string ciudad, string estado);

        Publicacion CrearTesis(string nombre, string[] nombres_completos_autores, int año, Clase_de_Tesis clase, string departamento, string institucion, string ciudad, string estado);
    }
}
